src/main/java/seedu/moneygowhere/currency/CurrencyApi.java

```diff
@@ -0,0 +1,106 @@
+package seedu.moneygowhere.currency;
+
+import seedu.moneygowhere.common.Configurations;
+import seedu.moneygowhere.common.Messages;
+import seedu.moneygowhere.data.currency.CurrencyManager;
+import seedu.moneygowhere.data.currency.Currency;
+import seedu.moneygowhere.exceptions.CurrencyLoadDataInputException;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.FileNotFoundException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+public class CurrencyApi {
+    public static void getCurrencyApi(CurrencyManager currencyManager) {
+        getJson();
+        loadFromFile(currencyManager);
+    }
+
+    public static File initialiseFile() {
+        File directory = new File(Configurations.DIRECTORY_PATH);
+        directory.mkdir();
+        String newFilePath = new File(Configurations.FILE_PATH_CURRENCIES).getAbsolutePath();
+        return new File(newFilePath);
+    }
+
+    private static Currency createCurrency(String textFromFile) throws CurrencyLoadDataInputException {
+        String[] splitInputs = textFromFile.split(":");
+        if (splitInputs.length != 2) {
+            throw new CurrencyLoadDataInputException();
+        }
+        String currencyCode = splitInputs[0];
+        Float rate = Float.parseFloat(splitInputs[1]);
```

Comment:
Do use BigDecimal instead of floats or doubles when representing currency amounts.